pkg/controller/common/autoscaling/validations.go

```diff
@@ -149,6 +149,28 @@ func ValidateAutoscalingPolicies(
 			)
 		}
 
+		// all roles should be valid
+		for _, role := range autoscalingSpec.Roles {
+			if !esv1.IsValidRole(role) {
+				errs = append(
+					errs,
+					field.Invalid(
+						autoscalingSpecPath(i, "name"), strings.Join(autoscalingSpec.Roles, ","),
+						fmt.Sprintf("node role %s is invalid", role)),
+				)
+			}
+		}
+
+		// at least ml or a data role is present
+		if !CanBeAutoscaled(autoscalingSpec.Roles) {
```

Comment:
I think I initially wanted to limit this kind of validation rules since it binds the life cycle of the Elasticsearch deciders with the life cycle of the operator itself. Said differently, if we follow this path, we might have to introduce rules depending on Elasticsearch versions when new deciders are introduced to allow them to be used. Instead, I would expect Elasticsearch to return an error if an autoscaling policy does not contain any role that can autoscaled.

Reply:
I see the point. In the other side, we already have plenty of es-feature version gates in the operator, which is one thing among others that makes it powerful. Once in place, it is something stable. Also, there shouldn't be many new deciders in the future.
I don't have a strong opinion but it seems to me that it's worth it to the user versus the cost to us, which seems to me pretty low.

Reply:
I tested and the UX is not so bad. I will see all the same to update the doc to remove the ambiguity.